aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1444,7 +1444,9 @@ Tensor nanmean(
 static Tensor& logsumexp_out_impl(Tensor& result, const Tensor& self, IntArrayRef dims, bool keepdim) {
   // can't take max of empty tensor
   if (self.numel() != 0) {
-    auto maxes = at::amax(self, dims, true);
+    // For complex numbers, use the real part to calculate the max. Based on
+    // https://scicomp.stackexchange.com/questions/34273/log-sum-exp-trick-for-signed-complex-numbers
+    auto maxes = at::amax(self.is_complex() ? at::real(self) : self, dims, true);
```

Comment:
nit: call tensor.real() on real tensors is a noop so this conditional isn't necessary.